server/src/api/auth.ts

```diff
@@ -77,11 +78,11 @@ router.post('/login', async (req, res) => {
   const { userId, password } = req.body;
   if (!(userId && password)) return res.sendStatus(400);
   let user;
-  [user] = await executeSql('select * from user where user_id = ?', [userId]);
+  [user] = (await executeSql('select * from user where user_id = ?', [userId])) as RowDataPacket[];
   if (!user) return res.status(401).json({ msg: '아이디 또는 비밀번호가 틀렸어요.' });
 
   const encrypted = encrypt(password, user.salt);
-  [user] = await executeSql('select * from user where user_id = ? and password = ?', [userId, encrypted]);
+  [user] = (await executeSql('select * from user where user_id = ? and password = ?', [userId, encrypted])) as RowDataPacket[];
```

Comment:
as ~~ 로 타입을 강제하는게 좋은 방법은 아니라고 하더라구요..
```suggestion
  [user] = (await executeSql('select * from user where user_id = ? and password = ?', [userId, encrypted]));
  if(!(user typeof RowDataPacket[])) { //이부분을 정확히는 모르겠지만
     ~~에러처리
  } ;
```

이런식으로 에러를 잡아주면 좋을것같습니다

Reply:
`mysql2` 내부 함수의 리턴 타입은 에러를 내지 않는 정상적인 상황에서도 **쿼리문에 따라** 달라집니다.
- `RowDataPacket[]`
- `RowDataPacket[][]`
- `OkPacket`
- `OkPacket[]`
- `ResultSetHeader`

쿼리문이 정해지는 순간 리턴 타입이 정해집니다. (이를 테면 `insert` 문을 실행할 경우 `OKPacket` 타입이 리턴)
쿼리문 작성과 타입 캐스팅이 같은 라인의 코드에서 이루어지므로 예상치 못한 문제가 발생할 가능성이 작다고 생각됩니다.

위 코드의 경우
DB 에러가 나지 않는 한, `user` 변수가 반드시 `RowDataPacket` 타입으로 강제되므로 타입체크의 이점을 얻을 수 있고,
DB 에러가 날 경우엔 catch되어 코드 실행이 되지 않습니다.

이러한 이유로 함수의 리턴 타입 자체를 체크해 분기하는 것은 불필요한 일이라고 생각됩니다.